.github/workflows/modular-arch-quality-gates.yml

```diff
@@ -0,0 +1,479 @@
+---
+# yamllint disable rule:line-length
+name: Modular Architecture - Quality Gates
+
+'on':
+  pull_request:
+    paths:
+      - 'frontend/packages/**'
+      - 'packages/**'  # Future path support
+  workflow_dispatch:
+    inputs:
+      modules:
+        description: 'Space-separated module names to run (optional)'
+        required: false
+        type: string
+
+permissions:
+  contents: read
+  actions: read
+
+env:
+  NODE_VERSION: '20'
+  GO_VERSION: '1.24.3'
+  # Configurable paths for future flexibility
+  PACKAGES_PATHS: 'frontend/packages,packages'
```

Comment:
we can depcreate `frontend/packages` I think everything is already migrated to `packages.`